tests/test_commons_auth.py

```diff
@@ -0,0 +1,166 @@
+        self.gcp_bundle_get_checkout_url = self.construct_url(f'/v1/bundles/checkout/{gcp_arn}', Replica.gcp)
+
+    def test_get_bundle_checkout_auth_errors(self):
+        self.run_auth_errors(self.aws_bundle_get_checkout_url, 'get')
+        self.run_auth_errors(self.gcp_bundle_get_checkout_url, 'get')
+
+    def test_post_bundle_checkout_auth_errors(self):
+        self.run_auth_errors(self.aws_bundle_post_checkout_url, 'post')
+        self.run_auth_errors(self.gcp_bundle_post_checkout_url, 'post')
+
+    def test_get_file_auth_errors(self):
+        self.run_auth_errors(self.aws_file_url, 'get')
+        self.run_auth_errors(self.gcp_file_url, 'get')
+
+    def test_put_file_auth_errors(self):
+        self.run_auth_errors(self.aws_file_url, 'put')
+        self.run_auth_errors(self.gcp_file_url, 'put')
+
+    def test_head_file_auth_errors(self):
+        self.run_auth_errors(self.aws_file_url, 'head')
+        self.run_auth_errors(self.gcp_file_url, 'head')
+
+    def test_get_bundle_auth_errors(self):
+        self.run_auth_errors(self.aws_bundle_url, 'get')
+        self.run_auth_errors(self.gcp_bundle_url, 'get')
+
+    def test_put_bundle_auth_errors(self):
+        self.run_auth_errors(self.aws_bundle_url, 'put')
+        self.run_auth_errors(self.gcp_bundle_url, 'put')
+
+    def test_delete_bundle_auth_errors(self):
+        self.run_auth_errors(self.aws_bundle_url, 'delete')
+        self.run_auth_errors(self.gcp_bundle_url, 'delete')
+
+    def test_post_search_auth_errors(self):
+        self.run_auth_errors(self.aws_search_url, 'post',
+                             error_key='x-amzn-ErrorType',
+                             error_value='UnauthorizedException',
+                             post_search=True)
+        self.run_auth_errors(self.gcp_search_url, 'post',
+                             error_key='x-amzn-ErrorType',
+                             error_value='UnauthorizedException',
+                             post_search=True)
+
+    def run_auth_errors(self, url, calltype,
+                        error_key='Content-Type',
+                        error_value='application/problem+json',
+                        post_search=False):
+        if calltype == 'get':
+            response = self.assertGetResponse
+        elif calltype == 'put':
+            response = self.assertPutResponse
+        elif calltype == 'delete':
+            response = self.assertDeleteResponse
+        elif calltype == 'head':
+            response = self.assertHeadResponse
+        elif calltype == 'post':
+            response = self.assertPostResponse
+
+        # Unauthorized email
+        with self.throw_403():
+            resp_obj = response(url, requests.codes.forbidden, headers=get_auth_header())
+            self.assertEqual(resp_obj.response.headers[error_key], error_value)
+
+        # Gibberish auth header
+        expected_code = requests.codes.unauthorized if not post_search else requests.codes.forbidden
```

Comment:
This line can now be removed:
```
expected_code = requests.codes.unauthorized if not post_search else requests.codes.forbidden
```
and `requests.codes.forbidden` can simply be passed to `response(...)`

Reply:
I'd like to argue for keeping the `expected_code = requests.codes.unauthorized` truncated version.  It's used in two places, and for me at least, it made it clearer.